Urban-Fun/View Controllers/HostViewController.m

```diff
@@ -38,7 +39,47 @@ - (void)viewDidLoad {
         [self.view addGestureRecognizer:tapGestureRecognizer];
     }
 
+- (IBAction)didTapImage:(id)sender {
+    UIImagePickerController *imagePickerVC = [UIImagePickerController new];
+    imagePickerVC.delegate = self;
+    imagePickerVC.allowsEditing = YES;
+    
+
+   
+    
+    UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"Upload Image"
+                                   message:@""
+                                   preferredStyle:UIAlertControllerStyleAlert];
+     
+    UIAlertAction* useCamera = [UIAlertAction actionWithTitle:@"Camera" style:UIAlertActionStyleDefault
+        handler:^(UIAlertAction * action){
+        if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
+            imagePickerVC.sourceType = UIImagePickerControllerSourceTypeCamera;
+            }
+        else {
+            imagePickerVC.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+        }
+        [self presentViewController:imagePickerVC animated:YES completion:nil];
+    }];
+    
+    UIAlertAction* usePhotoLibrary = [UIAlertAction actionWithTitle:@"Photo Library" style:UIAlertActionStyleDefault
+        handler:^(UIAlertAction * action){
+        imagePickerVC.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+        [self presentViewController:imagePickerVC animated:YES completion:nil];
+    }];
+    
+    
+    [alert addAction:useCamera];
+    [alert addAction:usePhotoLibrary];
+    [self presentViewController:alert animated:YES completion:nil];
+    
+}
 
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info {
+    UIImage *originalImage = info[UIImagePickerControllerOriginalImage];
+    self.activityImage.image = originalImage;
```

Comment:
Based on your code to open the `ImagePicker` you let the user edit the photo `imagePickerVC.allowsEditing = YES;`, in that case you should use the `UIImagePickerControllerEditedImage` to pick the edited one